silkrpc/core/evm_executor.cpp

```diff
@@ -167,27 +192,15 @@ asio::awaitable<ExecutionResult> EVMExecutor::call(const silkworm::Block& block,
 
                 const intx::uint256 base_fee_per_gas{evm.block().header.base_fee_per_gas.value_or(0)};
                 const intx::uint256 effective_gas_price{txn.effective_gas_price(base_fee_per_gas)};
-                const auto have = state.get_balance(*txn.from);
-                const auto want = txn.gas_limit * effective_gas_price;
-
+                auto want = txn.gas_limit * effective_gas_price;
                 const evmc_revision rev{evm.revision()};
-                if (rev >= EVMC_LONDON && txn.max_fee_per_gas < base_fee_per_gas) {
-                   silkworm::Bytes data{};
-                   std::string from = silkworm::to_hex(*txn.from);
-                   std::string error = "fee cap less than block base fee: address 0x" + from + ", gasFeeCap: " + intx::to_string(txn.max_fee_per_gas) + \
-                                         " baseFee: " + intx::to_string(base_fee_per_gas);
-                   ExecutionResult exec_result{1000, txn.gas_limit, data, error};
-                   asio::post(*context_.io_context, [exec_result, self = std::move(self)]() mutable {
-                      self.complete(exec_result);
-                   });
-                   return;
-                }
+                const intx::uint128 g0{silkworm::intrinsic_gas(txn, rev >= EVMC_HOMESTEAD, rev >= EVMC_ISTANBUL)};
+                assert(g0 <= UINT64_MAX); // true due to the precondition (transaction must be valid)
 
-                if (have < want) {
+                const auto error = pre_check(rev, txn, base_fee_per_gas, state, want, g0);
```

Comment:
Passing `silkworm::EVM` as parameter to `pre_check` allows removing both `rev` and `state` that can be obtained from `evm` as follows:
```
const evmc_revision rev{evm.revision()};
const silkworm::IntraBlockState state{evm.state()};
```

Reply:
ok